src/utils/HTTPUtils.ts

```diff
@@ -206,6 +208,24 @@ class HTTPUtils {
         // url is missing - returning options and callback
         return [options, callback];
     }
+
+    static obtainIncomingMessageEncoding(incomingMessage: http.IncomingMessage): string {
+        if (!incomingMessage || !incomingMessage.headers || !incomingMessage.rawHeaders) {
+            return;
+        }
+
+        let contentEncoding = incomingMessage.headers
+            && (incomingMessage.headers['content-encoding']);
+        if (!contentEncoding) {
+            const encodingIndex = incomingMessage.rawHeaders.indexOf('Content-Encoding');
```

Comment:
HTTP headers are case insensitive so in theory it could be sent as `content-encoding` and it is valid usage according to HTTP.

Reply:
You' re right, `incomingMessage.headers` are case insensitive (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers). Predefined header names in `incomingMessage.headers` (built-in http) are all in lower case. But in all header names in `incomingMessage.rawHeaders` look like camel case like 'Content-Encoding' (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding) .`incomingMessage.rawHeaders` is an string array which contains header names and values, For directly check Content-Encoding is exist, i used orginal header name of content encoding.